kroxylicious-kms-provider-aws-kms-test-support/src/main/java/io/kroxylicious/kms/provider/aws/kms/AbstractAwsKmsTestKmsFacade.java

```diff
@@ -252,15 +250,18 @@ private void checkForError(String key, URI uri, int statusCode, HttpResponse<byt
                 if (error != null && error.isNotFound()) {
                     throw new UnknownAliasException(key);
                 }
-                throw new IllegalStateException("unexpected response %s (AWS error: %s) for request: %s".formatted(response.statusCode(), error, uri));
+                throw new UnsupportedRequestException("unexpected response %s (AWS error: %s) for request: %s".formatted(response.statusCode(), error, uri));
```

Comment:
You don't want to treat all unexpected conditions as `UnsupportedRequestExceptions`.  That might be a lie which will mislead or mask errors.  If AWS starts throwing 500's at us from our rotate call, we want to know.

Trap the 501 case and throw an exception for it.  In all other cases .. IllegalStateException is okay.

Reply:
sorry, I'd misunderstood the code.